staging/src/kubevirt.io/client-go/generated/kubevirt/clientset/versioned/typed/core/v1/virtualmachineinstance_expansion.go

```diff
@@ -55,100 +62,306 @@ type VirtualMachineInstanceExpansion interface {
 	// TODO not implemented yet
-	return nil, nil
+	//  requires clientConfig
+	return nil, fmt.Errorf("SerialConsole is not implemented yet in generated client")
 }
 
 func (c *virtualMachineInstances) USBRedir(vmiName string) (StreamInterface, error) {
 	// TODO not implemented yet
-	return nil, nil
+	//  requires clientConfig
+	return nil, fmt.Errorf("USBRedir is not implemented yet in generated client")
 }
 
 func (c *virtualMachineInstances) VNC(name string) (StreamInterface, error) {
 	// TODO not implemented yet
-	return nil, nil
+	//  requires clientConfig
+	return nil, fmt.Errorf("VNC is not implemented yet in generated client")
 }
 
 func (c *virtualMachineInstances) Screenshot(ctx context.Context, name string, options *v1.ScreenshotOptions) ([]byte, error) {
-	// TODO not implemented yet
-	return nil, nil
+	moveCursor := "false"
+	if options.MoveCursor == true {
+		moveCursor = "true"
+	}
+	res := c.client.Get().
+		AbsPath(fmt.Sprintf(vmiSubresourceURL, v1.ApiStorageVersion)).
+		Namespace(c.ns).
+		Resource("virtualmachineinstances").
+		Name(name).
+		SubResource("vnc", "screenshot").
+		Param("moveCursor", moveCursor).
+		Do(ctx)
+
+	raw, err := res.Raw()
+	if err != nil {
+		return nil, res.Error()
+	}
+
+	return raw, nil
 }
 
 func (c *virtualMachineInstances) PortForward(name string, port int, protocol string) (StreamInterface, error) {
 	// TODO not implemented yet
-	return nil, nil
+	//  requires clientConfig
+	return nil, fmt.Errorf("PortForward is not implemented yet in generated client")
 }
 
 func (c *virtualMachineInstances) Pause(ctx context.Context, name string, pauseOptions *v1.PauseOptions) error {
-	// TODO not implemented yet
-	return nil
+	body, err := json.Marshal(pauseOptions)
+	if err != nil {
+		return err
+	}
+
+	return c.client.Put().
+		AbsPath(fmt.Sprintf(vmiSubresourceURL, v1.ApiStorageVersion)).
+		Namespace(c.ns).
+		Resource("virtualmachineinstances").
+		Name(name).
+		SubResource("pause").
+		Body(body).
+		Do(ctx).
+		Error()
 }
 
 func (c *virtualMachineInstances) Unpause(ctx context.Context, name string, unpauseOptions *v1.UnpauseOptions) error {
-	// TODO not implemented yet
-	return nil
+	body, err := json.Marshal(unpauseOptions)
+	if err != nil {
+		return err
+	}
+
+	return c.client.Put().
+		AbsPath(fmt.Sprintf(vmiSubresourceURL, v1.ApiStorageVersion)).
+		Namespace(c.ns).
+		Resource("virtualmachineinstances").
+		Name(name).
+		SubResource("unpause").
+		Body(body).
+		Do(ctx).
+		Error()
 }
 
 func (c *virtualMachineInstances) Freeze(ctx context.Context, name string, unfreezeTimeout time.Duration) error {
-	// TODO not implemented yet
-	return nil
+	log.Log.Infof("Freeze VMI %s", name)
+	freezeUnfreezeTimeout := &v1.FreezeUnfreezeTimeout{
+		UnfreezeTimeout: &metav1.Duration{
+			Duration: unfreezeTimeout,
+		},
+	}
+
+	body, err := json.Marshal(freezeUnfreezeTimeout)
+	if err != nil {
+		return err
+	}
+	if err != nil {
+		return err
+	}
```

Comment:
The second err check is redundant. Probably a leftover.

Reply:
Changed, PTAL Thanks